### PR TITLE
feat: file context support added

### PR DESCRIPTION
**Pull request by kisharnath**

### CODE REVIEW

File: crowdin_api/api_resources/source_files/resource.py [hunk 1]
```diff
@@ -294,6 +294,7 @@ def add_file(
         branchId: Optional[int] = None,
         directoryId: Optional[int] = None,
         title: Optional[int] = None,
+        context:Optional[str]=None,
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/source_files/resource.py:297]:
```suggestion
        context: Optional[str]=None,
```

**Review by andrii-bodnar — CHANGES_REQUESTED** [review]

File: crowdin_api/api_resources/source_files/tests/test_source_files_resources.py [hunk 2]
```diff
@@ -361,6 +361,7 @@ def test_list_files(self, m_request, in_params, request_params, base_absolut_url
                     "branchId": None,
                     "directoryId": None,
                     "title": None,
+                    "context":None,
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/source_files/tests/test_source_files_resources.py:364]:
```suggestion
                    "context": None,
```

File: crowdin_api/api_resources/source_files/tests/test_source_files_resources.py [hunk 3]
```diff
@@ -375,6 +376,7 @@ def test_list_files(self, m_request, in_params, request_params, base_absolut_url
                     "branchId": 1,
                     "directoryId": 1,
                     "title": "title",
+                     "context":"context",
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/source_files/tests/test_source_files_resources.py:379]:
```suggestion
                     "context": "context",
```

File: crowdin_api/api_resources/source_files/tests/test_source_files_resources.py [hunk 4]
```diff
@@ -387,6 +389,7 @@ def test_list_files(self, m_request, in_params, request_params, base_absolut_url
                     "branchId": 1,
                     "directoryId": 1,
                     "title": "title",
+                    "context":"context",
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/source_files/tests/test_source_files_resources.py:392]:
```suggestion
                    "context": "context",
```

File: crowdin_api/api_resources/source_files/tests/test_source_files_resources.py [hunk 5]
```diff
@@ -401,6 +404,7 @@ def test_list_files(self, m_request, in_params, request_params, base_absolut_url
...
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/source_files/tests/test_source_files_resources.py:407]:
```suggestion
                    "context": "context",
```

File: crowdin_api/api_resources/source_files/tests/test_source_files_resources.py [hunk 6]
```diff
@@ -413,6 +417,7 @@ def test_list_files(self, m_request, in_params, request_params, base_absolut_url
...
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/source_files/tests/test_source_files_resources.py:420]:
```suggestion
                    "context": "context",
```

File: crowdin_api/api_resources/source_files/tests/test_source_files_resources.py [hunk 7]
```diff
@@ -427,6 +432,7 @@ def test_list_files(self, m_request, in_params, request_params, base_absolut_url
...
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/source_files/tests/test_source_files_resources.py:435]:
```suggestion
                    "context": "context",
```

File: crowdin_api/api_resources/source_files/tests/test_source_files_resources.py [hunk 8]
```diff
@@ -439,6 +445,7 @@ def test_list_files(self, m_request, in_params, request_params, base_absolut_url
...
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/source_files/tests/test_source_files_resources.py:448]:
```suggestion
                    "context": "context",
```

File: crowdin_api/api_resources/source_files/tests/test_source_files_resources.py [hunk 9]
```diff
@@ -453,6 +460,7 @@ def test_list_files(self, m_request, in_params, request_params, base_absolut_url
...
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/source_files/tests/test_source_files_resources.py:463]:
```suggestion
                    "context": "context",
```

File: crowdin_api/api_resources/source_files/tests/test_source_files_resources.py [hunk 10]
```diff
@@ -465,6 +473,7 @@ def test_list_files(self, m_request, in_params, request_params, base_absolut_url
...
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/source_files/tests/test_source_files_resources.py:476]:
```suggestion
                    "context": "context",
```

**Review by andrii-bodnar — CHANGES_REQUESTED** [review]

**Comment by andrii-bodnar** [comment]
@kisharnath thank you! There are a lot of linter issues - https://github.com/crowdin/crowdin-api-client-python/actions/runs/6609696157/job/17950251205

could you please take a look?

**Comment by kisharnath** [comment]
@andrii-bodnar  Could you please check now?

**Comment by andrii-bodnar** [comment]
@kisharnath the lint passes now, but there are some failed tests - https://github.com/crowdin/crowdin-api-client-python/actions/runs/6610035511/job/17951221769

File: crowdin_api/api_resources/source_files/resource.py [hunk 11]
```diff
@@ -250,7 +250,7 @@ def edit_directory(
         )
 
     # Files
-    def get_file_path(self, projectId: int, fileId: Optional[int] = None):
+    def get_file_path(self, projectId: int, context: str, fileId: Optional[int] = None):
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/source_files/resource.py:253]: this change is not needed here

File: crowdin_api/api_resources/source_files/resource.py [hunk 12]
```diff
@@ -330,14 +331,15 @@ def add_file(
...
+                "context": context,
                 "importOptions": importOptions,
                 "exportOptions": exportOptions,
                 "excludedTargetLanguages": excludedTargetLanguages,
                 "attachLabelIds": attachLabelIds,
             },
         )
 
-    def get_file(self, projectId: int, fileId: int):
+    def get_file(self, projectId: int, context: str, fileId: int):
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/source_files/resource.py:342]: this method doesn't accept the context parameter

File: crowdin_api/api_resources/source_files/resource.py [hunk 13]
```diff
@@ -421,7 +424,7 @@ def delete_file(self, projectId: int, fileId: int):
             path=self.get_file_path(projectId=projectId, fileId=fileId),
         )
 
-    def edit_file(self, projectId: int, fileId: int, data: Iterable[FilePatchRequest]):
+    def edit_file(self, projectId: int, context: str, fileId: int, data: Iterable[FilePatchRequest]):
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/source_files/resource.py:427]: we should only update the `FilePatchRequest` here, the `edit_file` parameters shouldn't be chnged

**Comment by kisharnath** [comment]
Please check now

**Review by andrii-bodnar — APPROVED** [review]
@kisharnath thank you!

**Pull request merged**